src/fastga_he/powertrain_builder/powertrain.py

```diff
@@ -544,6 +546,243 @@ def _get_connections(self):
+                    1 for connection in connections_list if comp in connection.get("source")
+                )
+
+                if int(input_count_defined) != int(input_count):
+                    raise ValueError("Connector component is/are not properly connected!")
+
+        if one_to_many_component:
+            for comp, output_count in zip(one_to_many_component, one_to_many_output_count):
+                output_count_defined = 0
+
+                output_count_defined += sum(
+                    1 for connection in connections_list if comp in connection.get("target")
+                )
+
+                if int(output_count_defined) != int(output_count):
+                    raise ValueError("Connector component is/are not properly connected!")
+
+        if many_to_many_component:
+            for comp, input_count, output_count in zip(
+                many_to_many_component, many_to_many_input_count, many_to_many_output_count
+            ):
+                input_count_defined = 0
+                output_count_defined = 0
+
+                input_count_defined += sum(
+                    1 for connection in connections_list if comp in connection.get("source")
+                )
+
+                output_count_defined += sum(
+                    1 for connection in connections_list if comp in connection.get("target")
+                )
+
+                if (int(input_count_defined) != int(input_count)) or (
+                    int(output_count_defined) != int(output_count)
+                ):
+                    raise ValueError("Connector component is/are not properly connected!")
+
+    def _categorize_connectors(self, one_to_one_component):
+        connector_names = [
+            comp
+            for comp, type in zip(self._components_name, self._components_type_class)
+            if type == "connector"
+        ]
+        connector_options = [
+            option
+            for option, type in zip(self._components_options, self._components_type_class)
+            if type == "connector"
+        ]
+        connector_variable = [
+            input
+            for name in connector_names
+            for input in self._components_connection_inputs
+            if name in input
+        ] + [
+            output
+            for name in connector_names
+            for output in self._components_connection_outputs
+            if name in output
+        ]
+
+        one_to_many_component = []
+        many_to_one_component = []
+        many_to_many_component = []
+        many_to_one_input_count = []
+        one_to_many_output_count = []
+        many_to_many_input_count = []
+        many_to_many_output_count = []
+
+        for name, options in zip(connector_names, connector_options):
+            variable_list = [var for var in connector_variable if name in var]
+
+            # Check many-to-many component
+            if options is None or not any(key.startswith("number_of_") for key in options.keys()):
```

Comment:
If option is None then no options will start with "number_of_". I get you want compact things but here it becomes hard to read and more complex that it could be